dotnet-validate/Program.cs

```diff
@@ -52,35 +53,56 @@ private static async Task<int> Main(string[] args)
 
         private static async Task<int> RunLocalCommand(string file)
         {
-            try
+            var directory = Directory.GetCurrentDirectory();
+
+            if (Path.GetPathRoot(file) is { } root && !string.IsNullOrEmpty(root))
             {
-                using var cancellationTokenSource = new CancellationTokenSource();
-                Console.CancelKeyPress += (_, eventArgs) =>
-                {
-                    eventArgs.Cancel = !cancellationTokenSource.IsCancellationRequested;
-                    cancellationTokenSource.Cancel();
-                };
+                directory = root;
+                file = Path.GetRelativePath(root, file);
+            }
 
-                var packageFile = new FileInfo(file);
-                if (packageFile.Exists)
-                {
-                    await Console.Out.WriteLineAsync($"Validating {packageFile.FullName}").ConfigureAwait(false);
-                }
+            var files = new Matcher(StringComparison.Ordinal)
+                .AddInclude(file)
+                .Execute(new DirectoryInfoWrapper(new DirectoryInfo(directory)));
 
-                var isValid = await RunAsync(packageFile, cancellationTokenSource.Token).ConfigureAwait(false);
+            if (!files.HasMatches)
+                return EXIT_FAILURE;
 
-                return isValid ? EXIT_SUCCESS : EXIT_FAILURE;
-            }
-            catch (UnavailableException exception)
-            {
-                await Console.Error.WriteLineAsync(exception.Message).ConfigureAwait(false);
-                return EX_UNAVAILABLE;
-            }
-            catch (Exception exception)
+            foreach (var actualFile in files.Files)
             {
-                await Console.Error.WriteLineAsync(exception.ToString()).ConfigureAwait(false);
-                return EX_SOFTWARE;
+                try
+                {
+                    using var cancellationTokenSource = new CancellationTokenSource();
+                    Console.CancelKeyPress += (_, eventArgs) =>
```

Comment:
Can this event handler and the `cancellationTokenSource` be moved outside of the `foreach`? And then maybe add a `cancellationTokenSource.Token.IsCancellationRequested` check inside of the loop.

Reply:
Good point. Was completely oblivious about that cancellationtoken. Done.